flink-ml-core/src/main/java/org/apache/flink/ml/common/datastream/DataStreamUtils.java

```diff
@@ -113,32 +151,44 @@ public static <T> DataStream<T> reduce(DataStream<T> input, ReduceFunction<T> fu
             extends AbstractUdfStreamOperator<OUT, MapPartitionFunction<IN, OUT>>
             implements OneInputStreamOperator<IN, OUT>, BoundedOneInput {
 
-        private ListState<IN> valuesState;
+        private final TypeInformation<IN> inputType;
 
-        public MapPartitionOperator(MapPartitionFunction<IN, OUT> mapPartitionFunc) {
+        private DataCacheListState<IN> dataCacheListState;
+
+        public MapPartitionOperator(
+                MapPartitionFunction<IN, OUT> mapPartitionFunc, TypeInformation<IN> inputType) {
             super(mapPartitionFunc);
+            this.inputType = inputType;
```

Comment:
Instead of passing inputType to create serializer, would it be simpler to keep the previous approach and create serializer using the following code?

```
getOperatorConfig().getTypeSerializerIn(0, getClass().getClassLoader())
```